File: 2996 - Smallest Missing Integer Greater Than Sequential Prefix Sum/Method 1/solution.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var missingInteger = function(nums) {
    let map = new Map();
    for(let num of nums) {
         map.set(num, (map.get(num) || 0) + 1);
    }
    let sum = nums[0];
    for(let i = 1; i<nums.length; i++) {
        if(nums[i]==nums[i-1]+1){
            sum = sum + nums[i];
        } else {
            break;
        }
    }
    while(map.has(sum)) {
        sum++;
    }
    return sum;
};

File: 2996 - Smallest Missing Integer Greater Than Sequential Prefix Sum/Method 1/solution.java
class Solution {
    public int missingInteger(int[] nums) {
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int num : nums) {
            map.put(num, map.getOrDefault(num, 0)+1);
        }
        int sum = nums[0];
        for(int i = 1; i<nums.length; i++) {
            if(nums[i]==nums[i-1]+1) {
                sum += nums[i];
            } else {
                break;
            }
        }
        while(map.containsKey(sum)) {
            sum++;
        }
        return sum;
    }
}